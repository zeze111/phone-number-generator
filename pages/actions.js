import fetch from "isomorphic-unfetch";

export const generateNumbers = limit =>
  fetch("http://localhost:3000/telecomms/", {
    method: "POST",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json"
    },
    body: JSON.stringify({ limit })
  }).then(response => response.json());

export const listNumbers = (offset, limit) =>
  fetch(
    `http://localhost:3000/telecomms/phone-numbers?limit=${limit}&offset=${offset}`
  ).then(response => response.json());

export const sortNumbers = (order, offset, limit)  =>
  fetch(
    `http://localhost:3000/telecomms/phone-numbers/sort/${order}?limit=${limit}&offset=${offset}`
  ).then(response => response.json());

export const clearNumbersStorage = () =>
  fetch("http://localhost:3000/telecomms/phone-numbers", {
    method: "DELETE",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json"
    }
  }).then(response => response.json());
