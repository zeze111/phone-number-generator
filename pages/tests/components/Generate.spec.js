import React from "react";
import { shallow } from "enzyme";
import Generate from "../../components/generate";
import { generateNumbers } from "../../actions";

jest.mock("../../actions", () => ({
  generateNumbers: jest.fn(() => ({
    numbers: ["012", "123", "234", "345"]
  }))
}));

describe("Generate", () => {
  it("should render the component", () => {
    const wrapper = shallow(<Generate />);
    wrapper.setState({ numbers: ["012", "123", "234", "345"] });
    expect(wrapper).toMatchSnapshot();
  });

  describe("onChange", () => {
    it("should change event", () => {
      const event = {
        target: { name: "limit", value: "10" }
      };
      const wrapper = shallow(<Generate />);
      const limit = wrapper.find("#limit");
      limit.simulate("change", event);
      expect(wrapper.instance().state.limit).toBe("10");
    });
  });

  describe("onSubmit", () => {
    const event = {
      preventDefault: jest.fn()
    };
    const wrapper = shallow(<Generate />);

    wrapper.setState({ limit: "5" });
    wrapper.instance().onSubmit(event);
    expect(generateNumbers).toHaveBeenCalledWith("5");
  });
});
