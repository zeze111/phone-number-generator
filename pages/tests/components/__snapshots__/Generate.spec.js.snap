// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generate should render the component 1`] = `
<div
  className="generate-content"
>
  <h3>
     Generate New Numbers 
  </h3>
  <p
    className="generate-text"
  >
    Please enter the amount of numbers to generate
  </p>
  <form
    className="generate-form"
  >
    Limit:
    <input
      id="limit"
      name="limit"
      onChange={[Function]}
      type="text"
    />
    <button
      className="submit-button"
      onClick={[Function]}
      type="button"
    >
      Enter
    </button>
  </form>
  <div
    className="generate-numbers"
  >
    <ul>
      <li
        className="numbers"
        key="0"
      >
        012
      </li>
      <li
        className="numbers"
        key="1"
      >
        123
      </li>
      <li
        className="numbers"
        key="2"
      >
        234
      </li>
      <li
        className="numbers"
        key="3"
      >
        345
      </li>
    </ul>
  </div>
</div>
`;
